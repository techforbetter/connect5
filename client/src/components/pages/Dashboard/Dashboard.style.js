import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { colors, shadows, breakpoints } from '../../../theme';

export const Wrapper = styled.div`
  padding: 80px 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 750px;
  margin: 0 auto;
`;

export const TopSection = styled.div`
  margin-bottom: 1rem;
  width: 100%;
  @media ${breakpoints.mobileL} {
    text-align: center;
    margin: 1.2rem 0;
  }
`;

export const Title = styled.h2`
  font-size: 1.5rem;
  font-weight: 500;
  margin-bottom: 0.5rem;
`;

export const Role = styled.p`
  font-size: 14px;
  font-weight: 300;
  text-transform: capitalize;
  @media ${breakpoints.mobileL} {
    font-size: 1rem;
    font-weight: 300;
  }
`;

export const TotalReach = styled(Role)``;

export const StatsWrapper = styled.div`
  background-color: ${colors.white};
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
  box-shadow: ${shadows.primary};
  margin-bottom: 1rem;
`;

export const StatItem = styled(Link)`
  text-decoration: none;
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 1.5rem 0;
  color: ${colors.profileFontColor};

  :hover {
    color: ${colors.profileFontColor};
  }
  @media ${breakpoints.mobileL} {
    padding: 2rem 0;
  }
`;

export const Label = styled.h3`
  font-weight: 300;
  font-size: 16px;
  margin-bottom: 0.5rem;
`;

export const StatNumber = styled.p`
  font-size: 1.5rem;
  color: ${colors.lightPrimary};
`;

export const StyledLink = styled(Link)`
  font-weight: 300;
  font-size: 14px;
  color: ${colors.profileFontColor};
  padding: 0.5rem 0;

  :hover {
    color: ${colors.red};
  }
`;

export const LogOut = styled.div`
  font-weight: 300;
  font-size: 14px;
  color: ${colors.profileFontColor};
  padding: 0.5rem 0;
  cursor: pointer;
  background: none;
  border: none;

  :hover {
    color: ${colors.red};
  }
`;

export const SpinWrapper = styled.div`
  height: 100vh;
  width: 100vw;
`;
