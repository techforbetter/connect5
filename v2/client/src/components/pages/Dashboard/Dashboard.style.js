import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { colors, shadows } from '../../../theme';

export const Wrapper = styled.div`
  padding: 80px 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const TopSection = styled.div`
  margin-bottom: 2rem;
  width: 100%;
  @media (min-width: 678px) {
    width: 65%;
    margin: 0 atuo;
    text-algin: center;
  }
`;

export const Title = styled.h2`
  font-size: 1.5rem;
  font-weight: 500;
  margin-bottom: 0.5rem;
  @media (min-width: 678px) {
    text-align: center;
  }
`;

export const Role = styled.p`
  font-size: 14px;
  font-weight: 300;
  text-transform: capitalize;
  @media (min-width: 678px) {
    text-align: center;
  }
`;

export const StatsWrapper = styled.div`
  background-color: ${colors.white};
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
  box-shadow: ${shadows.primary};
  margin-bottom: 1rem;
  @media (min-width: 678px) {
    width: 65%;
    margin: 0 auto;
  }
`;

export const StatItem = styled(Link)`
  text-decoration: none;
  width: 50%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 1.5rem 0;
  color: ${colors.profileFontColor};

  :hover {
    color: ${colors.profileFontColor};
  }
`;

export const Label = styled.h3`
  font-weight: 300;
  font-size: 16px;
  margin-bottom: 0.5rem;
`;

export const StatNumber = styled.p`
  font-size: 1.5rem;
  color: ${colors.lightPrimary};
`;

export const StyledLink = styled(Link)`
  font-weight: 300;
  font-size: 14px;
  color: ${colors.profileFontColor};
  padding: 0.5rem 0;

  :hover {
    color: ${colors.red};
  }
`;

export const LogOut = styled.div`
  font-weight: 300;
  font-size: 14px;
  color: ${colors.profileFontColor};
  padding: 0.5rem 0;
  cursor: pointer;

  :hover {
    color: ${colors.red};
  }
`;

export const SpinWrapper = styled.div`
  height: 100vh;
  width: 100vw;
`;
