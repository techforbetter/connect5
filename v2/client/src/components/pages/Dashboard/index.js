import React, { Component } from 'react';
import { connect } from 'react-redux';

import Spin from 'antd/lib/spin';

import { fetchStatsData } from '../../../actions/users';

import {
  Wrapper,
  TopSection,
  Title,
  Role,
  StatsWrapper,
  StatItem,
  Label,
  StatNumber,
  StyledLink,
  SpinWrapper,
} from './Dashboard.style';

import Header from '../../common/Header';

import {
  TRAINER_RESULTS_URL,
  TRAINER_SESSIONS_URL,
  GROUP_RESULTS_URL,
  TRAINERS_URL,
  GROUP_SESSIONS_URL,
} from '../../../constants/navigationRoutes';

class Dashboard extends Component {
  componentDidMount() {
    // once we have log in sorted role and name will come from props
    // it can then be removed from state
    // const { role, userName } = this.props;
    // const role = 'trainer';
    const { role, fetchStatsData: fetchStatsDataActionCreator } = this.props;

    fetchStatsDataActionCreator(role);
    // this.setState({ userType });
    // const { fetchStatsData: fetchStatsDataActionCreator } = this.props;
    // fetchStatsDataActionCreator(userType);
  }

  render() {
    const { userName, stats } = this.props;
    // const { id } = auth;

    const captalizesName =
      userName && userName[0].toUpperCase() + userName.substr(1);

    const { role } = this.props;

    return (
      <Wrapper>
        <Header type="home" />
        {!stats.loaded ? (
          <SpinWrapper>
            <Spin size="large" />
          </SpinWrapper>
        ) : (
          <>
            <TopSection>
              <Title>
                Welcome back, <br />
                {captalizesName}
              </Title>
              <Role>Role: {role}</Role>
            </TopSection>
            <StatsWrapper>
              <StatItem
                to={
                  role === 'trainer' ? TRAINER_SESSIONS_URL : GROUP_SESSIONS_URL
                }
              >
                <Label>Sessions</Label>
                <StatNumber>{stats.sessionCount}</StatNumber>
              </StatItem>
              <StatItem
                to={
                  role === 'trainer' ? TRAINER_RESULTS_URL : GROUP_RESULTS_URL
                }
              >
                <Label>Participants</Label>
                <StatNumber>{stats.participantCount}</StatNumber>
              </StatItem>
              <StatItem
                to={
                  role === 'trainer' ? TRAINER_RESULTS_URL : GROUP_RESULTS_URL
                }
              >
                <Label>Responses</Label>
                <StatNumber>{stats.responseCount}</StatNumber>
              </StatItem>
              {role === 'trainer' ? (
                <StatItem to={TRAINER_RESULTS_URL}>
                  <Label>Response Rate</Label>
                  <StatNumber>{stats.responseRate}%</StatNumber>
                </StatItem>
              ) : (
                <StatItem to={TRAINERS_URL}>
                  <Label>Trainers</Label>
                  <StatNumber>{stats.trainerCount}</StatNumber>
                </StatItem>
              )}
            </StatsWrapper>
            <StyledLink to="/change-password">Change Password</StyledLink>
            <StyledLink to="/logout">Log out</StyledLink>
          </>
        )}
      </Wrapper>
    );
  }
}

const mapStateToProps = state => {
  return {
    userName: state.auth.name,
    stats: state.stats,
    userId: state.auth.id,
    role: state.auth.role,
  };
};

export default connect(
  mapStateToProps,
  { fetchStatsData }
)(Dashboard);
