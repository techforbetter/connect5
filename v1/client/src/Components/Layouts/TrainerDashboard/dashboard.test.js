import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import Dashboard from "./index";

test("Dashboard matches snapshot", () => {
  const context = {};

  const component = renderer.create(
    <StaticRouter context={context}>
      <Dashboard />
    </StaticRouter>,
  );
  expect(component).toMatchSnapshot();
});
