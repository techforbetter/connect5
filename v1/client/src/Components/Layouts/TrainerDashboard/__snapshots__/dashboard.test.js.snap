// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches snapshot 1`] = `
<h1>
  Loading your details...
</h1>
`;
